### ISSUE TITLE
Possible to use Regex patterns in Topic string?

### ISSUE THREAD

**Issue by joewood** [confluentinc/ksql#529]
The Kafka Client library allows subscription to topics as a regex pattern. Is it possible to use the same feature when creating a stream or table?

**Comment by miguno** [comment]
FYI (copied from Confluent public Slack discussion).

**Comment by miguno** [comment]
I suppose you are asking though: "If it isn't supported yet, can you please add this?" :-)

**Comment by joewood** [comment]
Yes please. We use streams with regex patterns today. If this is inline with the strategy I may give it a shot in my branch and send a PR.

**Comment by miguno** [comment]
> If this is inline with the strategy I may give it a shot in my branch and send a PR.

It's definitely in-line with the strategy!

The open question, which @hjafarpour was referring to when talking about 1-1 relationship between stream/table and topic, is how easily we can integrate it into / adapt the current data model of KSQL.

Let us know if you want to bounce off ideas etc. on how this could be achieved.

**Comment by joewood** [comment]
I was thinking of simply extending the `CREATE STREAM` DDL to include a `kafka_topic_patten` option:
```SQL
CREATE STREAM pageviews (viewtime BIGINT, user_id VARCHAR, page_id VARCHAR)
  WITH (VALUE_FORMAT = 'JSON',
        KAFKA_TOPIC_PATTERN = 'my\-pageviews[0-9]\-topic');
```

This would a single stream from multiple topics, using `builder.stream( Pattern.compile(xx))`. That seems like a quick fix PR, but I suspect that I'll also need to update the metastore based on the current set of available topics.

**Comment by miguno** [comment]
> but I suspect that I'll also need to update the metastore based on the current set of available topics

Yeah, I think implications like this is what @hjafarpour was hinting at.

**Comment by mjsax** [comment]
Using wildcard would not make sense for `TABLES` IMHO because you would end up with race conditions between the topics, as there are no ordering guarantees between partitions. Also note, that KafakStreams does not support wildcard-subscription for `builder.table(...)` the the same reason.

For `STREAMS` it might be possible, however, it would be difficult to define how `INSERT INTO` should work? -- Maybe we could support wildcards for "source streams" though, that are read only. (cf. KLIP-49: https://github.com/confluentinc/ksql/pull/7474)
